app/controllers/api/v1/providers_controller.rb

```diff
@@ -4,37 +4,47 @@ class ProvidersController < ApplicationController
       include NextLinkHeader
 
       # Potential edge case:
-      # It is possible for older updated_at values to written to the database after this API has been queried for changes. This would mean that these changes are missed when the client makes a subsequent request using the next-link.
+      #
+      # It is possible for older updated_at values to written to the database
+      # after this API has been queried for changes. This would mean that these
+      # changes are missed when the client makes a subsequent request using the
+      # next-link.
+      #
       # Possible causes of older updated_at values:
-      # - delay between c# calculating datetime.UtcNow and value being written to postgres
+      # - delay between c# calculating datetime.UtcNow and value being written
+      #   to postgres
```

Comment:
Any ideas of the lag?

Maybe move the time setting to the database...

Reply:
We haven't witnessed the lag happening, we just hypothesized that it's possible. We try to protect against it by acquiring a lock on the table to make sure that there isn't an update in-flight that might cause this (line 21 below).